### PR TITLE
Clean up authentication mess.

### LINKED ISSUE #142: Parameters are lost on reauthentication?
Noticed this weird behavior:

```
mreg> host find -name ThisDoesNotExist
You need to re-autenticate
Enter password: ******************************
WARNING: : Too many hits, 56388, more than limit of 500. Refine search.
mreg> host find -name ThisDoesNotExist
WARNING: : No hosts found.
```

Are the parameters somehow scrambled/lost when reauthenticating?

### LINKED ISSUE #160: Re-authentication interrupts action
### Issue:

When the mreg-token expires and an action is performed `mreg-cli` returns
```
WARNING: : GET "https://mreg.uio.no/api/v1/hosts/?page=2": 404: Not Found
{
  "detail": "Invalid page."
}
```
After re-authenticating again instead of fulfilling the request

### Steps to reproduce

1. Log into mreg-cli
2. Delete `.mreg-cli_auth_token`
3. Perform any action `host info bybanen`
4. Re-authenticate

### LINKED ISSUE #205: Incorrect authentication credentials doesn't prevent CLI from launching
```
mreg-cli --user pederhan --url https://mreg.uio.no --domain uio.no 
Connecting to https://mreg.uio.no
Password for pederhan: ********
ERROR: : {'detail': 'Incorrect authentication credentials.'}
Type -h for help.
mreg>
```
This used to raise an error and exit the application before #191 was merged.

On using a command, the application prompts again for new authentication information, so this isn't an _error_ per say, it's just divergent behavior from pre-#191.

### PR DESCRIPTION
**Pull request by terjekv**
- Cleans up the names for the auth functions.
  - Adds a LoginFailedError exception.
  - Passes on params from the pre-relogin command to the post-login command.

Fixes #205, fixes #160,  fixes #142

### CODE REVIEW

File: mreg_cli/exceptions.py [hunk 1]
```diff
@@ -58,3 +58,20 @@ class NetworkNotFoundWarning(CliWarning):
...
+    def formatted_exception(self) -> str:
+        """Return a string formatted with 'Login failed:' prefixing the error message.
+
+        :returns: Formatted error message.
+        """
+        return f"Login failed: {super().__str__()}"
+
+    def __str__(self) -> str:
+        """Return the error message."""
+        return "Login failed"
```
Comment (pederhan) [inline mreg_cli/exceptions.py:75]: Could this call `self.formatted_exception` instead for consistency?

```suggestion
        return self.formatted_exception()
```

File: mreg_cli/utilities/api.py [hunk 2]
```diff
@@ -80,18 +90,29 @@ def login1(user: str, url: str) -> None:
...
     print(f"Connecting to {url}")
-    # get url
     password = prompt(f"Password for {user}: ", is_password=True)
     try:
-        _update_token(user, password)
+        auth_and_update_token(user, password)
     except CliError as e:
-        e.print_self()
+        if catch_exception:
+            e.print_self()
```
Comment (pederhan) [inline mreg_cli/utilities/api.py:113]: Note: would be nice if we had unit tests that ensured we print something useful here.

I think it's fine for now, but this behavior is a bit opaque from the caller's perspective.

File: mreg_cli/exceptions.py (hunk 1 above)
Comment (terjekv) [reply mreg_cli/exceptions.py:75]: The problem (ish) is that I'm not sure what we should say/leak to the user when logging in fails. This is why I made the default `__str__` be completely vague.

File: mreg_cli/utilities/api.py (hunk 2 above)
Comment (terjekv) [reply mreg_cli/utilities/api.py:113]: Well, mreg-cli should have unit tests. :)

File: mreg_cli/exceptions.py (hunk 1 above)
Comment (pederhan) [reply mreg_cli/exceptions.py:75]: Sure. That makes a lot of sense from a security perspective.

**Review by pederhan — APPROVED** [review]
LGTM

**Pull request merged**